pkg/bmc/virtualmedia_validation.go

```diff
@@ -0,0 +1,100 @@
+package bmc
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// contains tells whether the array of redfish.VirtualMediaType
+// has the element redfish.VirtualMediaType.
+func contains(array []redfish.VirtualMediaType, element redfish.VirtualMediaType) bool {
+	for _, n := range array {
+		if element == n {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidateVirtualMedia returns nil when the BMC has support
+// for VirtualMedia and error when the BMC does not have support.
+func ValidateVirtualMedia(address string, user string, password string, insecure bool, httpClient *http.Client) error {
```

Comment:
This method is becoming pretty big, I'd evaluate the possibility to break it and refactor it into a dedicated struct type to simplify it

Reply:
ack I will take a look at it

Reply:
I did a refactor in the function